numpyro/contrib/reparam.py

```diff
@@ -55,6 +55,14 @@ def process_message(self, msg):
         new_fn, value = reparam(msg["name"], msg["fn"], msg["value"])
 
         if value is not None:
+            if new_fn is None:
+                msg['type'] = 'deterministic'
+                msg['value'] = value
+                for key in list(msg.keys()):
+                    if key not in ('type', 'name', 'value'):
+                        del msg[key]
+                return
```

Comment:
Returns `deterministic` site here instead of `sample` site. This way, MCMC can report results for deterministic nodes.

Reply:
How do we handle this in Pyro?

Reply:
In Pyro, we use Delta observation `sample` site because Pyro doesn't have `deterministic` site yet.

Reply:
I see. So with this change, we'll need to wrap this in `Predictive` to get the values at deterministic nodes still, right?

Reply:
For both Delta obs or deterministic, we need to use Predictive. But with deterministic, we can retrace the model at postprocess_fn, as we currently do, so no need to use Predictive. I think we need to do the same for autoguide but I'll address it separately.